cpp/tests/io/csv_test.cpp

```diff
@@ -2194,4 +2194,38 @@ TEST_F(CsvReaderTest, DtypesMapInvalid)
   EXPECT_THROW(cudf_io::read_csv(in_opts), cudf::logic_error);
 }
 
+TEST_F(CsvReaderTest, CsvDefaultOptionsWriteReadMatch)
+{
+  auto filepath = temp_env->get_temp_dir() + "issue.csv";
```

Comment:
generally speaking, `const` is preferably when possible. It reduces cognitive overhead (does this value every get reassigned? If it's const, we know it _can't_ and won't ever have to consider it changing).
```suggestion
  auto const filepath = temp_env->get_temp_dir() + "issue.csv";
```
const could be added to a few of the variable declarations below, as well.